Fix special queries in dataview queries

jcdeichmann opened the pull request:
Fixes #233 

## The Problem
Dataview queries were recently added as a database source. The query is stored in the database markdown file as a key value pair where the value is a string enclosed in double quotes. Dataview queries use double quotes  (`"`) when specifying folders, and was causing issues when saving to the markdown file since they weren't escaped. Additionally, newlines (`\n`) mess up the database markdown file since new key/values are expected on each line.

##  The Solution
To fix this, I've modified the  `UnmarshallConfigHandler` to escape double quotes and new lines on save, and the `MarshallConfigHandler` to unescape them when repopulating the settings from the markdown file. 

The below query has double quotes (`"`) and a new line (`\n`)
<img width="668" alt="image" src="https://user-images.githubusercontent.com/16976706/182353695-3b153788-93fe-428f-be19-557971e62c4f.png">

Here's the query successfully working
<img width="873" alt="image" src="https://user-images.githubusercontent.com/16976706/182353779-2be7709d-0184-400e-be5f-74f1fb2fe5df.png">

Here's the escaped query in the markdown file
<img width="325" alt="image" src="https://user-images.githubusercontent.com/16976706/182355882-e52906a8-8c2b-4101-a98c-c511596ac487.png">


This PR is ready for review, @RafaelGB

RafaelGB commented on src/parsers/handlers/marshall/MarshallConfigHandler.ts:
```diff
@@ -31,7 +32,11 @@ export class MarshallConfigHandler extends AbstractYamlHandler {
     loadDefaultConfig<K extends keyof LocalSettings>(key: K, value: Literal, localSettings: LocalSettings): LocalSettings {
-        localSettings[key] = value as any;
+        var escapedValue = value
+        if (typeof value === "string") {
```
Dataview has a WrappedLiteral where you can obtain the type. This should work too, but could be interesting to use it
(examples in dataviewService.ts)

RafaelGB replied:
also. Could you use `let`instead of `var`? rollup build optimisation

RafaelGB reviewed (approved):
Looks nice for me. Thanks for your contribution!

jcdeichmann commented:
> Looks nice for me. Thanks for your contribution!

Shoot -- I just tested a couple more things and realized I was testing a previous build. There's a couple bugs in this. I'll investigate and update in a bit -- sorry about that!

RafaelGB commented:
no rush, I just add a comment with the WrappedLiteral.

There are no conflict with the migration of the dispatcher

jcdeichmann commented on src/helpers/QueryHelper.ts:
```diff
@@ -4,5 +4,5 @@ import { DatabaseCore } from "helpers/Constants";
     return `TABLE ${Object.keys(columns)
         .filter((key) => !key.startsWith("__") && !key.endsWith("__"))
-        .join(",")},${DatabaseCore.DATAVIEW_FILE},${DatabaseCore.FRONTMATTER_KEY} ${fromQuery.replace(/['"]+/g, '')}`
+        .join(",")},${DatabaseCore.DATAVIEW_FILE},${DatabaseCore.FRONTMATTER_KEY} ${fromQuery}`
```
For the new escaping to work, we have to stop stripping out the quotes and double quotes

RafaelGB replied:
Let me know when you think its ready to merge

RafaelGB commented:
I will merge it before generate conflicts with another branch

jcdeichmann commented:
> I will merge it before generate conflicts with another branch

Sorry -- I was at work and wasn't able to respond and address your comments before you merged this. I opened another PR addressing those comments: https://github.com/RafaelGB/obsidian-db-folder/pull/237

RafaelGB commented:
Oh thank you so much.

Sorry me merging it before your confirmation. I was absorbed with the big changes of the dispatcher and I wanted to make sure I didn't break your changes before I got my hands on the marshalls.
